mithril-aggregator/src/dependency_injection/builder/mod.rs

```diff
@@ -147,8 +149,26 @@ pub struct DependenciesBuilder {
     /// Genesis signature verifier service.
     pub genesis_verifier: Option<Arc<ProtocolGenesisVerifier>>,
 
+    /// Mithril signer registration master service
+    pub mithril_signer_registration_master: Option<Arc<MithrilSignerRegistrationMaster>>,
+
+    /// Mithril signer registration slave service
+    pub mithril_signer_registration_slave: Option<Arc<MithrilSignerRegistrationSlave>>,
+
     /// Signer registerer service
-    pub mithril_registerer: Option<Arc<MithrilSignerRegisterer>>,
+    pub signer_registerer: Option<Arc<dyn SignerRegisterer>>,
+
+    /// Signer synchronizer service
+    pub signer_synchronizer: Option<Arc<dyn SignerSynchronizer>>,
+
+    /// Signer registration verifier
+    pub signer_registration_verifier: Option<Arc<dyn SignerRegistrationVerifier>>,
+
+    /// Signer registration round opener service
+    pub signer_registration_round_opener: Option<Arc<dyn SignerRegistrationRoundOpener>>,
+
+    /// Signer registration pruning task
+    pub signer_registration_pruning_task: Option<Arc<dyn EpochPruningTask>>,
```

Comment:
Would it be better to not store the pruning task in the builder but instead make `get_signer_registration_pruning_task` return the correct task source directly ? They should already be stored in another field of the builder anyway.

Reply:
Good idea, done :+1: